python/raytracing/hyperboloid_utilities.py

```diff
@@ -220,8 +220,27 @@ def cusp_view_matrix(v):
 
     RF = v[0].parent()
 
-    return matrix([[1,0,0,0],
-                   [0,1,0,0],
-                   [0,0,1,0],
-                   [0,0,0,1]],
-                  ring = RF)
+    v = 1.1*v
+    center = vector([1,0,0,0])
+    t = 1 + 1/(r13_dot(center,v))
+    
+    point = R13_normalise(t*v + (1-t)*center)
+    direction = point - center
+    direction = R13_normalise(direction + r13_dot(direction, point)*point)
+
+    print(point)
+    print(direction)
+
+    m = matrix([
+        direction,
+        point,
+        [0,-1,0,0],
+        [0,0,-1,0]
+        ])
+    switch = matrix([
+        [0,1,0,0],
+        [0,0,1,0],
+        [0,0,0,1],
+        [1,0,0,0]
+        ])
+    return (switch*m).transpose()
```

Comment:
The client code does SO(1,3)-orthonormalize the matrix for you... But I still think you want to orthonormalize here to have better control of what the orthonormalization does - that is, do it before multiplying with switch, so that it doesn't change the camera position and view direction. Also, we might need to check that the determinant is positive and flip the last row of the matrix if it isn't.

Reply:
I now actually wonder whether my orthonormalization is doing the right thing in terms of whether we want to orthonormalize the rows or the columns. We want to only change the length of the camera position vector and not its direction...

Reply:
I now actually wonder whether my orthonormalization is doing the right thing in terms of whether we want to orthonormalize the rows or the columns. We want to only change the length of the camera position vector and not its direction...